.circleci/config.yml

```diff
@@ -1,240 +1,312 @@
 
 workflows:
-  version: 2
-  build:
-    when:
-      not:
-        equal: [ scheduled_pipeline, << pipeline.trigger_source >> ]
+  build-test-deploy:
     jobs:
-      - check-compilation-warnings
-      - check-code-formatting
-      - tests-dotnet:
-          name: dotnet-3.1-nightly
-          influxdb-image: "quay.io/influxdb/influxdb:nightly"
-          dotnet-image: "mcr.microsoft.com/dotnet/core/sdk:3.1"
-      - tests-dotnet:
-          name: dotnet-3.1
-          dotnet-image: "mcr.microsoft.com/dotnet/core/sdk:3.1"
-      - tests-dotnet:
-          name: dotnet-5.0
+      # - check-code-formatting # TODO: uncomment
+      - build
+      - test-docker:
+          name: test-dotnet-3.1-docker-nightly
+          influxdb2-docker-image: quay.io/influxdb/influxdb:nightly
+          dotnet-docker-image: mcr.microsoft.com/dotnet/sdk:3.1
+          dotnet-target-framework: netcoreapp3.1
+          requires:
+            - build
+      - test-docker:
+          name: test-dotnet-3.1-docker
+          dotnet-docker-image: mcr.microsoft.com/dotnet/sdk:3.1
+          dotnet-target-framework: netcoreapp3.1
+          requires:
+            - build
+      - test-docker:
+          name: test-dotnet-5.0-docker
+          dotnet-docker-image: mcr.microsoft.com/dotnet/sdk:5.0
+          dotnet-target-framework: net5.0
+          requires:
+            - build
+      - test-docker:
+          name: test-dotnet-6.0-docker
+          dotnet-docker-image: mcr.microsoft.com/dotnet/sdk:6.0
+          dotnet-target-framework: net6.0
+          requires:
+            - build
+      - test-docker:
+          name: test-dotnet-7.0-docker
+          dotnet-docker-image: mcr.microsoft.com/dotnet/sdk:7.0
+          dotnet-target-framework: net7.0
+          requires:
+            - build
+      - test-docker:
+          name: test-dotnet-8.0-docker
+          dotnet-docker-image: mcr.microsoft.com/dotnet/sdk:8.0
+          dotnet-target-framework: net8.0
           code-coverage-report: true
-          dotnet-image: "mcr.microsoft.com/dotnet/sdk:5.0"
-      - tests-dotnet:
-          name: dotnet-6.0
-          dotnet-image: "mcr.microsoft.com/dotnet/sdk:6.0"
-      - tests-dotnet:
-          name: dotnet-7.0
-      - tests-windows:
-          name: dotnet-windows
+          requires:
+            - build
+      - test-windows:
+          name: test-dotnet-4.8-windows
+          dotnet-target-framework: net48
+          requires:
+            - build
+      - test-windows:
+          name: test-dotnet-6.0-windows
+          dotnet-target-framework: net6.0
+          requires:
+            - build
       - deploy-preview:
           requires:
-            - check-compilation-warnings
-            - check-code-formatting
-            - dotnet-3.1-nightly
-            - dotnet-3.1
-            - dotnet-5.0
-            - dotnet-6.0
-            - dotnet-7.0
-            - dotnet-windows
+            # - check-code-formatting # TODO: uncomment
```

Comment:
Code formatting check is currently commented as many new C# features (introduced after C# 8) are applied by ReSharper (eg. [file scoped namespaces](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/file-scoped-namespaces), [target-typed new expressions](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/target-typed-new)).

I see two options:
- A big commit that reformats all files,
- Override some unnecessary formatting rules (like file scoped namespaces).

Reply:
We can leave it commented out for now and, after merging this PR, create another one to address the formatting issues. Does this approach work for you?